### ISSUE TITLE
Long term enhancement: Remove/reduce javascript usage

### ISSUE THREAD

**Issue by chanan** [chanan/BlazorStrap#135]
This will not be easy, but it would be really nice to not have any javascript in the project. Where we use JS now:

* Adding classes to the body tag for models - No idea how to to do this without JS
* Popper - Should be possible without JS, just a lot of work
* Tooltip - Same as popper
* FocusElement - Is this possible without JS?
* Loading bootstrap CS - this is possible without js, but hard and / or will be ugly. One way to do it is to download the css ourselves using HttpClient and sticking it into a `style` tag - is that even a good idea?

**Comment by herbertmilhomme** [comment]
this is relevant to #196. If you let the project remain C# only, and have it compound to already existing javascript functions provided by bootstrap base service (which is mandatory for use of blazorstrap). There isnt a need to repeat what bootstrap is already doing... we dont need to remove how a modal (or any other component) is made and replace it with a .net-signalr alternative. We can just leave it to do it's thing, and focus solely on creating `.razor` files instead? 

**tldr;** is `blazorstrap.js` a necessity? why is `aria-...` removed from `.razor` bootstrap components and models?

**Comment by chucker** [comment]
>is blazorstrap.js a necessity?

I think the answer to that question is why this issue exists. ;-)

**Comment by herbertmilhomme** [comment]
@chucker lol. That's why it's at the very end in a "too long; didnt read..." at the bottom of a very wordy paragraph stating that it wasnt necessary. I am in agreement with Original Poster and offering more points to support case. 

I was trying to get a conversation where both parties list a response and say "i am suggesting we do this" and the other side responds with a "i think it's better if we dont because...", because i failed to see why it was done in the first place. 

I am totally missing why "I think the answer to that question is why this issue exists. ;-)" is a reply to my question to SEEK more information, where as original poster didnt open dialogue for feedback and just stated we should get rid of it without any pros or cons to decision. 

Please dont tag me after this reply. I am unsubscribing to further updates.

**Comment by chanan** [comment]
To explain a bit - BlazorStrap does not use %99 of bootstrap javascript. In fact, the current version of bootstrap uses JQuery (this will change next version of bootstrap). BlazorStrap doesnt not use JQuery. There are a few places we couldnt (or choose not) to avoid javascript. The parts that we do use javascript for are listed in my initial description of the issue. 

Ideally, blazorstrap wouldn't use any javascript at all, however as I listed in the description that might not be possible, but we will try to reduce javascript as much as possible.

**Comment by herbertmilhomme** [comment]
>  because i failed to see why it was done in the first place.
@chanan addressing that subject would definitely do more to answer the question of "is blazorstrap.js a necessity?" You technically didn't answer anything in your explanation, so i missed the point of what the explanation was supposed to be expressing.

If there is no one to advocate for the other end of the argument then i guess it really is a moot point to hold further discussions in an echo chamber.

>Ideally, blazorstrap wouldn't use any javascript at all, however as I listed in the description that might not be possible, but we will try to reduce javascript as much as possible.

I am totally missing the purpose of this issue ticket... even if it is only marked as an `enhancement`, i dont see what it's enhancing, and why or how your solution does anything to help.

Something exist, and is there because it's needed to exist... but we're opening a ticket to talk about discussion of removing it, because we dont want it to exist?... Maybe it's one of those... "Let's do everything in the color red, because i like the color red" type of conversations, and probably why i cant see any logic in what's going on.
